### PR TITLE
Dedicated Assert event

### PR DESCRIPTION
**Pull request by ThomasHaas**
This PR adds a dedicated assert event rather than exploiting locals.

### CODE REVIEW

File: dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java [hunk 1]
```diff
@@ -248,15 +248,16 @@ public static String generateResultSummary(VerificationTask task, ProverEnvironm
                 printWarningIfThreadStartFailed(p, encCtx, prover);
                 if (props.contains(PROGRAM_SPEC) && FALSE.equals(model.evaluate(PROGRAM_SPEC.getSMTVariable(encCtx)))) {
                     summary.append("===== Program specification violation found =====\n");
-                    for(Event e : p.getThreadEvents(Local.class)) {
-                        if(e.hasTag(Tag.ASSERTION) && TRUE.equals(model.evaluate(encCtx.execution(e)))) {
+                    for(Assert ass: p.getThreadEvents(Assert.class)) {
+                        if(FALSE.equals(model.evaluate(encCtx.encodeExpressionAsBooleanAt(ass.getExpression(), ass)))) {
```
Comment (hernanponcedeleon) [inline dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java:252]: Why false? is the condition in the new assert event reverted wrt locals?

Comment (ThomasHaas) [reply dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java:252]: The original code checked for the assertion being executed not for its condition. So the old code implicitly assumed that we only ever have `assert(false)` (I think some pieces of code still have this assumption).

**Comment by hernanponcedeleon** [comment]
LGTM

**Comment by ThomasHaas** [comment]
I would still like to get the CI running before merging. I guess we can just retarget this PR to `development`, let the CI run, and then retarget it back.

**Comment by hernanponcedeleon** [comment]
That seems like more work than running the unit tests locally on you machine 😝

**Comment by ThomasHaas** [comment]
Merged into #493

**Pull request merged**